### ISSUE TITLE
Variable in the configuration file

### ISSUE THREAD

**Issue by yramamurthy** [pfefferf/ui5-nwabap-deployer#7]
How do I refer to an environment variable in the ui5.yaml file? I wanted to avoid hard coding of user credentials in the example configuration file provided below. The idea is to have the environment variables set for user/password and to refer to these environment variables in the configuration file. Appreciate any inputs. Thank you.

```
specVersion: '1.0'
metadata:
  name: ui5proj1
type: application
builder:
  customTasks:
  - name: ui5-task-nwabap-deployer
    afterTask: generateVersionInfo
    configuration: 
      resources:
        pattern: "**/*.*"
      connection:
        server: https://host:port
        client: 100
        useStrictSSL: false 
      authentication:
        user: xxx
        password: xxx
```

**Comment by pfefferf** [comment]
You cannot set an env variable in the UI5.yaml file. It needs to be set in the environment which uses the UI5 tooling. For instance if you execute the UI5 tooling in a shell, set the env variables there. 
As described in the documentation of that module, for development reasons the dotenv package is supported. So you can just drop a .env file with the variables in your project root (consider: just for dev reasons; avoid that for production or to commit .env files with sensitive information to your repo).

**Comment by yramamurthy** [comment]
@pfefferf my question is NOT about **setting** an env variable from the UI5.yaml file but reference it within the file after setting in the environment.

**Comment by pfefferf** [comment]
@yramamurthy You cannot reference the env variables in the file. They must be considered in the task implementation (which is done for the this task). Just check the documentation regarding the env names, set it, and the task will consider them.

**Comment by nxthor** [comment]
After a long search for documentation I was able to provide all personal data (credentials, transport number etc. by using the following environment variables in a `.env` file (project root):

```
UI5_TASK_NWABAP_DEPLOYER__SERVER=http://sap-system.host:port
UI5_TASK_NWABAP_DEPLOYER__CLIENT=yourClient
UI5_TASK_NWABAP_DEPLOYER__USER=yourUser
UI5_TASK_NWABAP_DEPLOYER__PASSWORD=yourPassword
UI5_TASK_NWABAP_DEPLOYER__TRANSPORTNO=yourCurrentTransportNo
```

Edit:
(no need to reference them in the ui5.yaml, the task will use them)
/Edit

I found the variable in the sourcecode of the deployer-task `ui5-nwabap-deployer` which is using the `ui5-nwabap-deployer-core`.

**Comment by pfefferf** [comment]
@nxthor What do you want to say us with your comment?

As usual all variables are documented in the README (https://github.com/pfefferf/ui5-nwabap-deployer/blob/master/packages/ui5-task-nwabap-deployer/README.md). Did I miss something?

**Comment by nxthor** [comment]
I posted this because I believe that this was the answer the OP/Author searched/asked for.

In terms of documentation: well I did not find it, maybe I googled it wrong, but I found this issue first.
I guess the OP had the same problem.
Also, I searched more in a context of a ui5-tooling feature, and was not aware of a dedicated repository (since a coworker introduced the task together with the tooling).
